src/main/java/com/gluonhq/substrate/target/AbstractTargetConfiguration.java

```diff
@@ -126,67 +125,60 @@ public boolean compile(String cp) throws IOException, InterruptedException {
+            compileRunner.addArg("-H:Log=registerResource:");
         }
-        compileBuilder.command().addAll(getTargetSpecificAOTCompileFlags());
+        compileRunner.addArgs(getTargetSpecificAOTCompileFlags());
         if (!getBundlesList().isEmpty()) {
             String bundles = String.join(",", getBundlesList());
-            compileBuilder.command().add("-H:IncludeResourceBundles=" + bundles);
+            compileRunner.addArg("-H:IncludeResourceBundles=" + bundles);
         }
-        compileBuilder.command().add("-Dsvm.platform=org.graalvm.nativeimage.Platform$"+jniPlatform);
-        compileBuilder.command().add("-cp");
-        compileBuilder.command().add(cp);
-        compileBuilder.command().addAll(projectConfiguration.getCompilerArgs());
-        compileBuilder.command().add(mainClassName);
+        compileRunner.addArg("-Dsvm.platform=org.graalvm.nativeimage.Platform$"+jniPlatform);
+        compileRunner.addArg("-cp");
+        compileRunner.addArg(cp);
+        compileRunner.addArgs(projectConfiguration.getCompilerArgs());
+        compileRunner.addArg(mainClassName);
 
-        postProcessCompilerArguments(compileBuilder.command());
+        postProcessCompilerArguments(compileRunner.getCmdList());
 
-        Logger.logDebug("compile command: " + String.join(" ", compileBuilder.command()));
+        compileRunner.setInfo(true);
+        compileRunner.setLogToFile(true);
         Path workDir = gvmPath.resolve(projectConfiguration.getAppName());
-        compileBuilder.directory(workDir.toFile());
-        compileBuilder.redirectErrorStream(true);
-        Process compileProcess = compileBuilder.start();
-        InputStream inputStream = compileProcess.getInputStream();
-        asynPrintFromInputStream(inputStream);
-        int result = compileProcess.waitFor();
-        // we will print the output of the process only if we don't have the resulting objectfile
+        int result = compileRunner.runProcess("compile", workDir.toFile());
 
         boolean failure = result != 0;
         String extraMessage = null;
         if (!failure) {
-            String nameSearch = mainClassName.toLowerCase() + "." + getObjectFileExtension();
+            // we will print the output of the process only if we don't have the resulting objectfile
+            String nameSearch = mainClassName.toLowerCase(Locale.ROOT) + "." + getObjectFileExtension();
             if (FileOps.findFile(gvmPath, nameSearch).isEmpty()) {
                 failure = true;
                 extraMessage = "Objectfile should be called "+nameSearch+" but we didn't find that under "+gvmPath.toString();
             }
         }
         if (failure) {
-            System.err.println("Compilation failed with result = " + result);
-            printFromInputStream(inputStream);
-
-            if (extraMessage!= null) {
-                System.err.println("Additional information: "+extraMessage);
+            if (extraMessage != null) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://app.codacy.com/manual/gluonhq/substrate/pullRequest?prid=4983430)